src/packagedcode/pypi_setup_py.py

```diff
@@ -0,0 +1,208 @@
+#
+# Copyright (c) Gram and others.
+# This code is copied and modified from dephell_setuptools https://github.com/pypa/setuptools
+# SPDX-License-Identifier: MIT
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+
+import ast
+from copy import deepcopy
+from pathlib import Path
+
+"""
+Parse setup.py files.
+"""
+
+# https://setuptools.readthedocs.io/en/latest/setuptools.html#metadata
+FIELDS = {
+    'author_email',
+    'author',
+    'classifiers',
+    'dependency_links',
+    'description',
+    'download_url',
+    'extras_require',
+    'install_requires',
+    'keywords',
+    'license_file',
+    'license',
+    'long_description_content_type',
+    'long_description',
+    'maintainer_email',
+    'maintainer',
+    'metadata_version',
+    'name',
+    'obsoletes',
+    'package_dir',
+    'platforms',
+    'project_urls',
+    'provides',
+    'python_requires',
+    'requires',
+    'setup_requires',
+    'tests_require',
+    'url',
+    'version',
+}
+
+
+def is_setup_call(element):
```

Comment:
Just wondering if the `is_setup_call` function from `packagedcode/pypi.py` can be used here.

Reply:
I actually created and copied is_setup_call() from packagedcode/pypi.py
They very closely related: I think the key difference is that this version works from a flat list of elements while the version in pypi.py works from a tree.
I added a note and a docstring. We could also remote the older code that I kept in pypi for now.